### PR TITLE
Cleanup some used by Cortex methods in compact.go

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <yb532204897@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

File: pkg/compact/compact.go [hunk 1]
```diff
@@ -394,58 +394,11 @@ func (cg *Group) Add(meta *metadata.Meta) error {
...
-	cg.mtx.Lock()
-	defer cg.mtx.Unlock()
-
-	max := int64(0)
-	for _, b := range cg.blocks {
-		if b.MaxTime < max {
-			max = b.MaxTime
-		}
-	}
-	return max
```
Comment (yeya24) [inline pkg/compact/compact.go:436]: This always returns 0.

File: pkg/compact/compact.go [hunk 2]
```diff
@@ -394,58 +394,11 @@ func (cg *Group) Add(meta *metadata.Meta) error {
...
-	cg.mtx.Lock()
-	defer cg.mtx.Unlock()
-
-	min := int64(0)
-	for _, b := range cg.blocks {
-		if b.MinTime < min {
-			min = b.MinTime
-		}
-	}
-	return min
```
Comment (yeya24) [inline pkg/compact/compact.go:422]: This always returns 0

**Review by pracucci — CHANGES_REQUESTED** [review]
Would you mind keeping these functions, please? We're leveraging on it in Cortex / Grafana Metrics Enterprise.

**Comment by bwplotka** [comment]
Yup, we need to think about external users of our packages.

**Comment by yeya24** [comment]
Thanks, I will close this pr. My question is that the `MaxTime` and `MinTime` methods seem wrong. Doesn't they always return 0?

File: pkg/compact/compact.go (hunk 2 above)
Comment (bwplotka) [reply pkg/compact/compact.go:422]: You are right - Let's fix it. `min := int64(0)` should be `min := math.MaxInt64`

File: pkg/compact/compact.go (hunk 1 above)
Comment (bwplotka) [reply pkg/compact/compact.go:436]: You are right - Let's fix it. max := int64(0) should be max := math.MinInt64 Also it should be ` b.MaxTime > max`

I don't think it was well implemented so wonder how those were used....  cc @pracucci 

Let's add tests as well? (:

**Review by bwplotka — COMMENTED** [review]
Well yea, it's totally bugged, let's at least fix it. Thanks for spotting this! :hugs:

File: pkg/compact/compact.go (hunk 1 above)
Comment (pracucci) [reply pkg/compact/compact.go:436]: They're definitely wrong 🤔  I will double check it where we use them. Thanks!

**Comment by yeya24** [comment]
OK. Let me fix them in this pr.

**Review by prmsrswt — APPROVED** [review]

**Review by pracucci — APPROVED** [review]
Thanks!

**Review by bwplotka — APPROVED** [review]
Thanks!

**Pull request merged**